http://sourceware.org/ml/crossgcc/2005-05/msg00165.html
Fixes a MIPS build problem (unrelated to NPTL)

Message-ID: <428E8B24.1000201@realitydiluted.com>
Date: Fri, 20 May 2005 20:13:08 -0500
From: "Steven J dot  Hill" <sjhill at realitydiluted dot com>
To:  crossgcc at sources dot redhat dot com,  toolchain at gentoo dot org, 
 Shay_Gal-On at pmc-sierra dot com,  TheNop at gmx dot net
Subject: New NPTL patches for crosstools and MIPS NPTL patches....

Greetings.

I have uploaded the latest NPTL patch for crosstool-0.34. I have also
uploaded a tarball of the patches necessary to build a MIPS NPTL
cross toolchain. To build a MIPS NPTL toolchain you will need the
released version of binutils-2.16 and the absolute latest GCC and
glibc code from the HEAD of cvs. Use the 'demo-mips-nptl.sh' script
to build the toolchain. Please report bugs or issues to the crossgcc
mailing list. Here is the link off of my FTP site:

    ftp://ftp.realitydiluted.com/crosstools/crosstool-0.34/

[Note: BOOTSTRAP_GCC is set by crosstool.sh when invoking make install-headers]

diff -ur glibc-2.4.0-20050406/sysdeps/mips/mips32/Makefile glibc-2.4.0-20050406-patched/sysdeps/mips/mips32/Makefile
--- glibc-2.4.0-20050406/sysdeps/mips/mips32/Makefile	2003-03-29 02:15:28.000000000 -0600
+++ glibc-2.4.0-20050406-patched/sysdeps/mips/mips32/Makefile	2005-04-12 21:36:51.318837655 -0500
@@ -1,3 +1,7 @@
+ifeq ($(filter -DBOOTSTRAP_GCC,$(CFLAGS)),)
 ifeq ($(filter -mabi=32,$(CC)),)
 CC += -mabi=32
 endif
+else
+CC += -D"_MIPS_SZPTR=32"
+endif

Signed-off-by: Robert P. J. Day <rpjday@mindspring.com>
